### PR TITLE
Backend paddle: Support deeponet and other examples

### PR DESCRIPTION
**Pull request by lijialin03**
1. Update deeponet of paddle to support new examples
2. Update operator examples as pinn_forward/Helmholtz_Dirichlet_2d_HPO.py style

### CODE REVIEW

File: deepxde/nn/paddle/deeponet.py [hunk 1]
```diff
@@ -75,8 +75,12 @@ def forward(self, inputs):
             raise AssertionError(
                 "Output sizes of branch net and trunk net do not match."
             )
-        x = paddle.einsum("bi,bi->b", x_func, x_loc)  # [batch_size, ]
-        x = paddle.reshape(x, [-1, 1])  # reshape [batch_size, ] to [batch_size, 1]
+        # Use the following formula to temporarily replace paddle.einsum()
+        # Because no higher(>=2) orderderivatives for the op now
+        x = paddle.sum(x_func*x_loc,axis = 1,keepdim=True)
```
Comment (lululxvi) [inline deepxde/nn/paddle/deeponet.py:80]: Reformat via black.

File: examples/operator/advection_aligned_pideeponet.py [hunk 2]
```diff
@@ -1,8 +1,19 @@
-"""Backend supported: tensorflow.compat.v1, tensorflow"""
+"""Backend supported: tensorflow.compat.v1, tensorflow, paddle"""
 import deepxde as dde
 import matplotlib.pyplot as plt
 import numpy as np
-from deepxde.backend import tf
+
+if dde.backend.backend_name == "paddle":
+    dim_x = 5
+    sin = dde.backend.paddle.sin
```
Comment (lululxvi) [inline examples/operator/advection_aligned_pideeponet.py:8]: Don't use `dde.backend`.

import paddle and then use `sin = paddle.sin`

Comment (lijialin03) [reply examples/operator/advection_aligned_pideeponet.py]: Can I use code like "if dde.backend.backend_name == "paddle":" to determine which backend the user is using? Or is it better to comment out some code？

Comment (lululxvi) [reply examples/operator/advection_aligned_pideeponet.py]: Yes, you can use it.

File: deepxde/icbc/boundary_conditions.py [hunk 3]
```diff
@@ -193,9 +193,9 @@ def __init__(
         self.batch_size = batch_size
 
         if batch_size is not None:  # batch iterator and state
-            if backend_name != "pytorch":
+            if backend_name not in ["pytorch","paddle"] :
```
Comment (lululxvi) [inline deepxde/icbc/boundary_conditions.py:196]: Format via black.

File: deepxde/icbc/boundary_conditions.py [hunk 4]
```diff
@@ -193,13 +191,11 @@ def __init__(
         self.batch_size = batch_size
 
         if batch_size is not None:  # batch iterator and state
-            if backend_name != "pytorch":
+            if backend_name not in ["pytorch", "paddle"]:
                 raise RuntimeError(
-                    "batch_size only implemented for pytorch backend"
+                    "batch_size only implemented for pytorch and paddle backend"
```
Comment (lululxvi) [inline deepxde/icbc/boundary_conditions.py:196]: Need to modify the docstring above.

**Comment by lululxvi** [comment]
Split this PR into different ones for DeepOnet and `batch_size` separately.

**Comment by lijialin03** [comment]
> Split this PR into different ones for DeepOnet and `batch_size` separately.
batch_size part has been splited in pr https://github.com/lululxvi/deepxde/pull/1424

File: deepxde/nn/paddle/deeponet.py [hunk 5]
```diff
@@ -75,8 +74,12 @@ def forward(self, inputs):
             raise AssertionError(
                 "Output sizes of branch net and trunk net do not match."
             )
-        x = paddle.einsum("bi,bi->b", x_func, x_loc)  # [batch_size, ]
-        x = paddle.reshape(x, [-1, 1])  # reshape [batch_size, ] to [batch_size, 1]
+        # Use the following formula to temporarily replace paddle.einsum()
+        # Because no higher(>=2) orderderivatives for the op now
+        x = paddle.sum(x_func * x_loc, axis=1, keepdim=True)
```
Comment (lululxvi) [inline deepxde/nn/paddle/deeponet.py:79]: It seems that `einsum` and this new one work exactly the same. If so, the comments can be removed.

File: deepxde/nn/paddle/deeponet.py [hunk 6]
```diff
@@ -143,7 +145,11 @@ def forward(self, inputs):
             raise AssertionError(
                 "Output sizes of branch net and trunk net do not match."
             )
-        x = paddle.einsum("bi,ni->bn", x_func, x_loc)
+        # Use the following formula to temporarily replace paddle.einsum()
+        # Because no higher(>=2) orderderivatives for the op now
+        x = x_func @ x_loc.T
```
Comment (lululxvi) [inline deepxde/nn/paddle/deeponet.py:150]: The same for this one.

File: deepxde/nn/paddle/deeponet.py [hunk 7]
```diff
@@ -75,8 +74,10 @@ def forward(self, inputs):
             raise AssertionError(
                 "Output sizes of branch net and trunk net do not match."
             )
-        x = paddle.einsum("bi,bi->b", x_func, x_loc)  # [batch_size, ]
-        x = paddle.reshape(x, [-1, 1])  # reshape [batch_size, ] to [batch_size, 1]
+        x = paddle.sum(x_func * x_loc, axis=1, keepdim=True)
+        # TODO:
```
Comment (lululxvi) [inline deepxde/nn/paddle/deeponet.py:78]: It seems the `TODO` can be removed, as `paddle.sum` works the same as `einsum`, so there is no need to use `einsum` anymore.

File: deepxde/nn/paddle/deeponet.py [hunk 8]
```diff
@@ -143,7 +143,9 @@ def forward(self, inputs):
             raise AssertionError(
                 "Output sizes of branch net and trunk net do not match."
             )
-        x = paddle.einsum("bi,ni->bn", x_func, x_loc)
+        x = x_func @ x_loc.T
+        # TODO:
```
Comment (lululxvi) [inline deepxde/nn/paddle/deeponet.py:147]: The same for this TODO.

File: examples/operator/advection_aligned_pideeponet.py [hunk 9]
```diff
@@ -1,8 +1,23 @@
...
+
+if dde.backend.backend_name == "paddle":
+    import paddle
+
+    dim_x = 5
+    sin = paddle.sin
+    cos = paddle.cos
+    concat = paddle.concat
+else:
+    import tf
```
Comment (lululxvi) [inline examples/operator/advection_aligned_pideeponet.py:14]: `from deepxde.backend import tf`

**Pull request merged**